### PR TITLE
Add class to trait visitor.

### PR DESCRIPTION
**Pull request by bergice**
This lets us replace classes that extend certain other classes with a provided list of traits.

### CODE REVIEW

File: src/UpgradeRule/PHP/ClassToTraitRule.php [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+use SilverStripe\Upgrader\CodeCollection\ItemInterface;
+use SilverStripe\Upgrader\UpgradeRule\PHP\Visitor\ClassToTraitVisitor;
+use SilverStripe\Upgrader\UpgradeRule\PHP\Visitor\ParentConnector;
+use SilverStripe\Upgrader\Util\MutableSource;
+
+/**
+ * Replaces occurrences of class extensions with traits.
+ *
+ * Class ClassToTraitRule
+ * @package SilverStripe\Upgrader\UpgradeRule\PHP
```
Comment (maxime-rainville) [inline src/UpgradeRule/PHP/ClassToTraitRule.php]: Let's avoid using `@package`

Comment (robbieaverill) [reply src/UpgradeRule/PHP/ClassToTraitRule.php]: This whole (autogenerated from PHPStorm) doc block is pointless, I'd suggest disabling it in your IDE templates

File: tests/UpgradeRule/PHP/fixtures/class-to-traits.testfixture [hunk 2]
```diff
@@ -0,0 +1,21 @@
+{
```
Comment (maxime-rainville) [inline tests/UpgradeRule/PHP/fixtures/class-to-traits.testfixture]: Can we add some more complex example e.g.: an abstract class with some implement clauses and some methods.

**Comment by maxime-rainville** [comment]
Travis build failure is just some linting issue.

**Pull request merged**